src-java/nbworker-topology/nbworker-storm-topology/src/main/java/org/openkilda/wfm/topology/nbworker/services/FlowOperationsService.java

```diff
@@ -557,6 +574,27 @@ private void validateFlow(FlowPatch flowPatch, Flow flow) {
             throw new IllegalArgumentException("Can not turn on ignore bandwidth flag and strict bandwidth flag "
                     + "at the same time");
         }
+
+        if ((flow.getVlanStatistics() != null && !flow.getVlanStatistics().isEmpty())
+                || (flowPatch.getVlanStatistics() != null && !flowPatch.getVlanStatistics().isEmpty())) {
+            boolean zeroResultSrcVlan = isResultingVlanValueIsZero(flowPatch.getSource(), flow);
+            boolean zeroResultDstVlan = isResultingVlanValueIsZero(flowPatch.getDestination(), flow);
+
+            if (!zeroResultSrcVlan && !zeroResultDstVlan) {
+                throw new IllegalArgumentException("To collect vlan statistics you need to set source or "
+                        + "destination vlan_id to zero");
+            }
+        }
+    }
+
+    private boolean isResultingVlanValueIsZero(PatchEndpoint patchEndpoint, Flow flow) {
+        boolean isResultVlanIsZero = flow.getSrcVlan() == 0;
+        Integer patchSrcVlanResult = Optional.ofNullable(patchEndpoint)
```

Comment:
why name of variable is `src`? you also use this method for dst endpoint

Reply:
Removed from the name